app/[locale]/[[...path]]/page.tsx

```diff
@@ -0,0 +1,119 @@
+import { notFound } from 'next/navigation';
+import { unstable_setRequestLocale } from 'next-intl/server';
+import type { FC } from 'react';
+
+import { setClientContext } from '@/client-context';
+import { MDXRenderer } from '@/components/mdxRenderer';
+import { WithLayout } from '@/components/withLayout';
+import { DEFAULT_VIEWPORT, ENABLE_STATIC_EXPORT } from '@/next.constants.mjs';
+import { dynamicRouter } from '@/next.dynamic.mjs';
+import { availableLocaleCodes, defaultLocale } from '@/next.locales.mjs';
+import { MatterProvider } from '@/providers/matterProvider';
+
+type DynamicStaticPaths = { path: string[]; locale: string };
+type DynamicParams = { params: DynamicStaticPaths };
+
+// This is the default Viewport Metadata
+// @see https://nextjs.org/docs/app/api-reference/functions/generate-viewport
+export const viewport = DEFAULT_VIEWPORT;
+
+// This generates each page's HTML Metadata
+// @see https://nextjs.org/docs/app/api-reference/functions/generate-metadata
+export const generateMetadata = async (c: DynamicParams) => {
+  const { path = [], locale = defaultLocale.code } = c.params;
```

Comment:
```suggestion
export const generateMetadata = async ({params} : DynamicParams) => {
  const { path = [], locale = defaultLocale.code } = params;
```

`c` doesn't mean much, we can deconstruct directly